Gems/Atom/Feature/Common/Assets/Materials/Types/StandardPBR.azsli

```diff
@@ -8,6 +8,8 @@
 
 #pragma once
 
+#define STANDARDPBR_MATERIAL_TYPE 1
```

Comment:
same as above.

Reply:
We will still need this because they are sharing the same lighting model, but the material parameters inside it are different, so in a full-screen draw call we won't be able to distinguish them without this macro, unless we put standard pbr and standard multylayer pbr into different lighting model, which is a more impactful change.

Reply:
There must be a more generic way to identify the different material types. Those are just the stock, common material types that come with the renderer. New material types can always be added with the same lighting models, whether they are created by hand or using material canvas. Material canvas currently has two “output nodes” that effectively mirror base PBR and standard PBR stock material types. They share the same lighting models and base code but all of the customization, options, and properties are generated by the graph. New output nodes can be added to mirror eye, skin, enhanced PBR, UI, and others created for special purposes. Standard multi-layer PBR was originally created for a specific project, and intended to be deprecated with the intention of blending layers being done via material canvas, or a more general purpose material blending solution.

Reply:
If all material types share the same lighting model, which means that all shaders have to use the same set of MaterialSrg, this will limits the developers to create their own shader (consider someone writing a shader needs to access 
a parameter only exists in one material type's MaterialSrg, that would fail, and the shader writer won't know this unless they see the error)

With such a macro to identify the different material types, the access of MaterialSrg in a common shader shared by multiple material types can be clearly separated into different part.

Yes it is better to find a more generic way to identify different material types in the shader, but I can not figure out a better way... Any suggestions?